pkg/kubelet/cm/dra/state/state_checkpoint_test.go

```diff
@@ -49,23 +49,69 @@ func TestCheckpointGetOrCreate(t *testing.T) {
 			[]ClaimInfoState{},
 		},
 		{
-			"Restore valid checkpoint",
+			"Restore checkpoint - single claim",
 			`{"version":"v1","entries":[{"DriverName":"test-driver.cdi.k8s.io","ClaimUID":"067798be-454e-4be4-9047-1aa06aea63f7","ClaimName":"example","Namespace":"default","PodUIDs":{"139cdb46-f989-4f17-9561-ca10cfb509a6":{}},"CDIDevices":{"test-driver.cdi.k8s.io":["example.com/example=cdi-example"]}}],"checksum":1988120167}`,
 			"",
-			[]ClaimInfoState{{
-				DriverName: "test-driver.cdi.k8s.io",
-				ClaimUID:   "067798be-454e-4be4-9047-1aa06aea63f7",
-				ClaimName:  "example",
-				Namespace:  "default",
-				PodUIDs:    sets.New("139cdb46-f989-4f17-9561-ca10cfb509a6"),
-				CDIDevices: map[string][]string{
-					"test-driver.cdi.k8s.io": {"example.com/example=cdi-example"},
+			[]ClaimInfoState{
+				{
+					DriverName: "test-driver.cdi.k8s.io",
+					ClaimUID:   "067798be-454e-4be4-9047-1aa06aea63f7",
+					ClaimName:  "example",
+					Namespace:  "default",
+					PodUIDs:    sets.New("139cdb46-f989-4f17-9561-ca10cfb509a6"),
+					CDIDevices: map[string][]string{
+						"test-driver.cdi.k8s.io": {"example.com/example=cdi-example"},
+					},
 				},
 			},
+		},
+		{
+			"Restore checkpoint - single claim - multiple devices",
+			`{"version":"v1","entries":[{"DriverName":"meta-test-driver.cdi.k8s.io","ClaimUID":"067798be-454e-4be4-9047-1aa06aea63f7","ClaimName":"example","Namespace":"default","PodUIDs":{"139cdb46-f989-4f17-9561-ca10cfb509a6":{}},"CDIDevices":{"test-driver-1.cdi.k8s.io":["example-1.com/example-1=cdi-example-1"],"test-driver-2.cdi.k8s.io":["example-2.com/example-2=cdi-example-2"]}}],"checksum":2113538068}`,
+			"",
+			[]ClaimInfoState{
+				{
+					DriverName: "meta-test-driver.cdi.k8s.io",
+					ClaimUID:   "067798be-454e-4be4-9047-1aa06aea63f7",
+					ClaimName:  "example",
+					Namespace:  "default",
+					PodUIDs:    sets.New("139cdb46-f989-4f17-9561-ca10cfb509a6"),
+					CDIDevices: map[string][]string{
+						"test-driver-1.cdi.k8s.io": {"example-1.com/example-1=cdi-example-1"},
+						"test-driver-2.cdi.k8s.io": {"example-2.com/example-2=cdi-example-2"},
+					},
+				},
+			},
+		},
+		{
+			"Restore checkpoint - multiple claims",
```

Comment:
I'm pretty sure this is planned already, but let's make sure we will have a test restoring multiple claims with multiple devices each.

Reply:
I plan to add that when the kubelet can actually generate / handle that case itself (in an upcoming PR for the v1alpha2 change).

Reply:
Fine for me.